plone/app/contenttypes/behaviors/viewlets.py

```diff
@@ -8,4 +10,27 @@ class LeadImageViewlet(ViewletBase):
 
     def update(self):
         self.context = ILeadImage(self.context)
-        self.available = True if self.context.image else False
+        self.available = self.is_visible
+
+    @property
+    def is_visible(self):
+        visible = False
+        if self.context.image:
+            is_visible = api.portal.get_registry_record(
+                'is_visible',
+                interface=ILeadImageSettings)
+            if is_visible:
+                visible = True
+        return visible
+
+    @property
+    def scale_name(self):
+        return api.portal.get_registry_record(
+                'scale_name',
+                interface=ILeadImageSettings)
+
+    @property
+    def css_class(self):
+        prefix = self.context.portal_type.split(' ')[0].lower()
+        name = "{0}Image".format(prefix)
```

Comment:
I don't the purpose of prefix. E.g. for portal_type "News Item" would result in "newsImage". For portal_type "News Folder" it would result in the same css_class.